common.eai

```diff
@@ -12831,7 +12832,6 @@ function SingleMeleeAttackAM takes boolean needs_exp, boolean has_siege, boolean
       endif
 
       if GetHeroXP(hero_unit[1]) < GetRandomInt(50,150) or GetHeroXP(hero_unit[1]) > GetRandomInt(300,850) or GetHeroXP(hero_unit[2]) > 450 or FoodUsed() > GetRandomInt(40,60) then
-
         set hall = ChooseAttackTarget(hall)
         if hall != null and UnitAlive(hall) then
```

Comment:
I'm not sure this does anything? ChooseAttackTarget will return null if it can't find a valid unit meaning the ali_target whatever it is will be ignored. So yes the variable isn't reset before using it but it can't be used unless the target was valid in the first place. And if it is valid target then ali_target will be set anyway.

Reply:
I'll merge anyway though.